src/App.js

```diff
@@ -0,0 +1,142 @@
+  setPageSize,
+  setTodosList,
+  page,
+  todos,
+  page_size,
+  todos_list,
+  error,
+  search_text,
+  setSearch,
+  fetchSearchedTodos,
+}) => {
+  const columns = [
+    { field: "id", title: "ID", align: "left", width: 200 },
+    {
+      field: "title",
+      title: "TITLE",
+      align: "left",
+      width: 400,
+    },
+    {
+      field: "completed",
+      title: "COMPLETED",
+      align: "left",
+      width: 200,
+    },
+  ];
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
+
+  useEffect(() => {
+    setTodosList(todos, page, page_size);
+  }, [page, page_size, todos]);
+
+  return (
+    <div className="App" data-test="App">
+      {error ? (
+        <span data-test="Error">Something Went Wrong!</span>
+      ) : (
+        <React.Fragment>
+          <div className="SearchField" data-test="SearchField">
+            <SearchField
+              placeholder={"Search by Title..."}
+              onChange={(value) => setSearch(value)}
+              value={search_text}
+              onSubmit={() => {
+                setPage(0);
+                setPageSize(0);
+                fetchSearchedTodos(search_text);
+              }}
+            />
+          </div>
+
+          <div>
+            {todos_list
+              ? todos_list.length > 0 && (
+                  <div className="paginator-container">
+                    <Paginator
+                      data-test="Paginator"
+                      hasPageSizeSelector
+                      pageSizeOptions={[10, 25, 50]}
+                      totalCount={todos ? todos.length : 0}
+                      SingleSelect={{
+                        DropMenu: { direction: "down" },
+                      }}
+                      selectedPageIndex={page}
+                      selectedPageSizeIndex={page_size}
+                      onPageSelect={(pageSelected) => setPage(pageSelected)}
+                      onPageSizeSelect={(pageSizeSelected) => {
+                        setPageSize(pageSizeSelected);
+                        setPage(0);
+                      }}
+                    />
+                  </div>
+                )
+              : null}
+            <DataTable data-test="DataTable" data={todos_list} minRows={0}>
+              <EmptyStateWrapper>
+                <Title>Todos searched not found.</Title>
+              </EmptyStateWrapper>
+              {todos_list
+                ? todos_list.length > 0 &&
+                  columns.map((column, i) => (
+                    <DataTable.Column
+                      key={i}
+                      field={column.field}
+                      width={column.width}
+                      align={column.align}
+                    >
+                      {column.title}
+                    </DataTable.Column>
+                  ))
+                : null}
+            </DataTable>
+          </div>
+        </React.Fragment>
+      )}
+    </div>
+  );
+};
+
+const mapStateToProps = (store) => {
```

Comment:
separate mapStateToProps and mapDispatchToProps to a different file and make this App component used within that connectedComponent.